include/pagmo/problems/minlp_rastrigin.hpp

```diff
@@ -0,0 +1,248 @@
+     *
+     * @return the lower and upper bounds for each of the decision vector components
+     */
+    std::pair<vector_double, vector_double> get_bounds() const
+    {
+        vector_double lb(m_dim_c + m_dim_i, -5.12);
+        vector_double ub(m_dim_c + m_dim_i, 5.12);
+        for (decltype(m_dim_i) i = m_dim_c; i < m_dim_i + m_dim_c; ++i) {
+            lb[i] = -10;
+            ub[i] = -5;
+        }
+        return {lb, ub};
+    }
+
+    /// Integer dimension
+    /**
+     * It returns the integer dimension of the problem.
+     *
+     * @return the integer dimension of the problem.
+     */
+    vector_double::size_type get_nix() const
+    {
+        return m_dim_i;
+    }
+    /// Gradients
+    /**
+     *
+     * It returns the fitness gradient for this UDP.
+     *
+     * The gradient is represented in a sparse form as required by
+     * problem::gradient().
+     *
+     * @param x the decision vector.
+     *
+     * @return the gradient of the fitness function
+     */
+    vector_double gradient(const vector_double &x) const
+    {
+        auto n = x.size();
+        vector_double g(n);
+        const auto omega = 2. * pagmo::detail::pi();
+        for (decltype(n) i = 0u; i < n; ++i) {
+            g[i] = 2 * x[i] + 10.0 * omega * std::sin(omega * x[i]);
+        }
+        return g;
+    }
+
+    /// Hessians
+    /**
+     *
+     * It returns the hessians for this UDP.
+     *
+     * The hessians are represented in a sparse form as required by
+     * problem::hessians().
+     *
+     * @param x the decision vector.
+     *
+     * @return the hessians of the fitness function
+     */
+    std::vector<vector_double> hessians(const vector_double &x) const
+    {
+        auto n = x.size();
+        vector_double h(n);
+        const auto omega = 2. * pagmo::detail::pi();
+        for (decltype(n) i = 0u; i < n; ++i) {
+            h[i] = 2 + 10.0 * omega * omega * std::cos(omega * x[i]);
+        }
+        return {h};
+    }
+
+    /// Hessians sparsity (only the diagonal elements are non zero)
+    /**
+     *
+     * It returns the hessian sparisty structure for this UDP.
+     *
+     * The hessian sparisty is represented in the form required by
+     * problem::hessians_sparsity().
+     *
+     * @return the hessians of the fitness function
+     */
+    std::vector<sparsity_pattern> hessians_sparsity() const
+    {
+        sparsity_pattern hs;
+        auto n = m_dim_c + m_dim_i;
+        for (decltype(n) i = 0u; i < n; ++i) {
+            hs.push_back({i, i});
+        }
+        return {hs};
+    }
+    /// Problem name
+    /**
+     *
+     *
+     * @return a string containing the problem name
+     */
+    std::string get_name() const
+    {
+        return "MINLP Rastrigin Function";
+    }
+    /// Extra informations
+    /**
+     *
+     * @return a string containing extra informations on the problem
+     */
+    std::string get_extra_info() const
+    {
+        std::ostringstream ss;
```

Comment:
In general I prefer the use of stringstream syntax over the string concatenation as it reads more clearly (after format) and allows for the immediate use of our printing functionalities. In this case, though, there is no need and its here for consistency with my own way of doing it :)